compiler/rustc_metadata/src/native_libs.rs

```diff
@@ -396,4 +384,55 @@ impl Collector<'tcx> {
             }
         }
     }
+
+    fn i686_arg_list_size(&self, item: &hir::ForeignItemRef<'_>) -> usize {
+        let argument_types: &List<Ty<'_>> = self.tcx.erase_late_bound_regions(
+            self.tcx
+                .type_of(item.id.def_id)
+                .fn_sig(self.tcx)
+                .inputs()
+                .map_bound(|slice| self.tcx.mk_type_list(slice.iter())),
+        );
+
+        argument_types.iter().fold(0, |accum: usize, ty: Ty<'_>| {
```

Comment:
It would be simpler to use `Iterator::sum` https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.sum instead of `fold` here.